### PR TITLE
feature(integration): Add dbt cloud quickstart

### PR DESCRIPTION
**Pull request by lgodin-nr**
# Summary

New quickstart for dbt Cloud. 

[Dbt](https://www.getdbt.com/) is a data transformation tool that comes in both cloud and open source versions. It is used by many companies in analytics and data engineering. This quickstart walks through the process of installing, configuring, and running an Airflow DAG that collects metadata from dbt Cloud and sends it to NR1. The quickstart includes a dashboard and alerts. 

The github repo is publicly available, but the documentation PR has not been merged yet. We have a commitment to have it published by Thursday June, 27. Can this quickstart be published at the same time? 

## Pre merge checklist

- [x] Did you check you NRQL syntax? - Does it work?
- [x] Did you include a Data source and Documentation reference?
- [ ] Are all documentation links publicly accessible?
- [x] Did you check your descriptive content for [voice and tone](https://docs.newrelic.com/docs/style-guide/writing-strategies/voice-strategies-docs-sound-new-relic/)?
- [x] Did you check your descriptive content for spelling and grammar errors?
- [x] Did you review your content with a subject matter expert? (e.g. a Browser agent quickstart is reviewed with a member of the Browser Agent team)

### Dashboards

- [x] Does the PR contain a screenshot for each of your dashboards?
- [x] Do your screenshots show data?
- [x] Has the [sanitization script](https://github.com/newrelic/newrelic-quickstarts/blob/main/CONTRIBUTING.md#dashboards) been run on each dashboard?

### Alerts

- [x] Did you check that your alerts actually work?

### CODE REVIEW

File: alert-policies/dbt-cloud/dbt-failed-test-row.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
+# Name of the alert
+name: Dbt Failed Test Row 
+
+# Description and details
+description: |+
```
Comment (sjyothi54) [inline alert-policies/dbt-cloud/dbt-failed-test-row.yml:5]: This description should be less than 240 characters

File: alert-policies/dbt-cloud/dbt-job-failed.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
+# Name of the alert
+name: Dbt Job Failed
+
+# Description and details
+description: |+
```
Comment (sjyothi54) [inline alert-policies/dbt-cloud/dbt-job-failed.yml:5]: This description should be less than 240 characters

File: alert-policies/dbt-cloud/dbt-resource-failed.yml [hunk 3]
```diff
@@ -0,0 +1,75 @@
+# Name of the alert
+name: Dbt Resource Failed 
+
+# Description and details
+description: |+
+  Generates critical alerts for dbt resources fail with an error. Dbt resources include models, snapshots, seeds and tests.
```
Comment (sjyothi54) [inline alert-policies/dbt-cloud/dbt-resource-failed.yml:6]: This description should be less than 240 characters

File: alert-policies/dbt-cloud/dbt-test-warning.yml [hunk 4]
```diff
@@ -0,0 +1,75 @@
+# Name of the alert
+name: Dbt Test Warning 
+
+# Description and details
+description: |+
```
Comment (sjyothi54) [inline alert-policies/dbt-cloud/dbt-test-warning.yml:5]: This description should be less than 240 characters

File: data-sources/dbt-cloud/config.yml [hunk 5]
```diff
@@ -0,0 +1,21 @@
+id: dbt-cloud 
+displayName: dbt Cloud 
+description: |
+  Integrates dbt Cloud with New Relic using an Airflow DAG. With our dashboard and alerts, you can easily find problems
+  with dbt Cloud runs and resources (models, seeds, snapshots and tests). Also includes the ability to collect failed 
+  test rows from dbt tests run in Snowflake. 
+icon: logo.svg
```
Comment (sjyothi54) [inline data-sources/dbt-cloud/config.yml:7]:
```suggestion
icon: logo.png
```

File: quickstarts/dbt-cloud/config.yml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+documentation:
+  - name: Installation docs
+    url:  https://github.com/newrelic-experimental/newrelic-dbt-cloud-integration/
+    description: Installation documentation for dbt Cloud observability using an Airflow DAG
+  - name: Git repository 
+    url: https://github.com/newrelic-experimental/newrelic-dbt-cloud-integration
+    description: Git repository for the Airflow DAG and additional documentation
+
+# Content / Design
+icon: logo.svg
```
Comment (sjyothi54) [inline quickstarts/dbt-cloud/config.yml:52]:
```suggestion
icon: logo.png
```

File: quickstarts/dbt-cloud/config.yml [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+dataSourceIds:
+  - dbt-cloud  
+
+# Reference to alert policies to be included in this quickstart
+alertPolicies:
+  - dbt-failed-test-row 
+  - dbt-job-failed
+  - dbt-resource-failed
+  - dbt-test-warning
```
Comment (sjyothi54) [inline quickstarts/dbt-cloud/config.yml:36]:
```suggestion
  - dbt-cloud
```

File: quickstarts/dbt-cloud/config.yml [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+  - dbt-test-warning 
+
+# Reference to dashboards to be included in this quickstart
+dashboards:
+  - dbt-cloud 
+
+# Documentation references
+documentation:
+  - name: Installation docs
+    url:  https://github.com/newrelic-experimental/newrelic-dbt-cloud-integration/
```
Comment (sjyothi54) [inline quickstarts/dbt-cloud/config.yml:45]:
```suggestion
    url:  https://docs.newrelic.com/docs/more-integrations/open-source-telemetry-integrations/dbt-cloud/dbt-cloud-integration/
```

File: quickstarts/dbt-cloud/config.yml [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+# Support level: New Relic | Verified | Community (required)
+level: Community
+
+# Authors of the quickstart (required)
+authors:
+  - Leo Godin 
+
+# Keywords for filtering / searching criteria in the UI
+keywords:
+  - guided install
```
Comment (sjyothi54) [inline quickstarts/dbt-cloud/config.yml:25]:
```suggestion
  - dbt cloud
```

File: quickstarts/dbt-cloud/config.yml [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+  - open source monitoring
```
Comment (sjyothi54) [inline quickstarts/dbt-cloud/config.yml:26]:
```suggestion
  - dbt
```

File: quickstarts/dbt-cloud/config.yml [hunk 11]
```diff
@@ -0,0 +1,53 @@
+# Sets the URL name of the quickstart on public I/O (required)
+slug: dbt-cloud-integration 
+
+# Displayed in the UI (required)
+title: Dbt Cloud Integration
```
Comment (sjyothi54) [inline quickstarts/dbt-cloud/config.yml:5]:
```suggestion
title: dbt Cloud Integration 
```

File: data-sources/dbt-cloud/config.yml [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+install:
+  primary:
+    link:
+      url: https://docs.newrelic.com/docs/more-integrations/open-source-telemetry-integrations/dbt-cloud/dbt-cloud-integration/
+keywords:
+  - open source monitoring
+  - guided install
```
Comment (sjyothi54) [inline data-sources/dbt-cloud/config.yml:14]:
```suggestion
  - dbt cloud
  - dbt
```

**Review by sjyothi54 — CHANGES_REQUESTED** [review]
@lgodin-nr Please make the necessary changes and make sure to use `dbt` throughout the integration.

File: alert-policies/dbt-cloud/dbt-failed-test-row.yml (hunk 1 above)
Comment (lgodin-nr) [reply alert-policies/dbt-cloud/dbt-failed-test-row.yml:5]: Updated

File: alert-policies/dbt-cloud/dbt-job-failed.yml (hunk 2 above)
Comment (lgodin-nr) [reply alert-policies/dbt-cloud/dbt-job-failed.yml:5]: Updated

File: alert-policies/dbt-cloud/dbt-test-warning.yml (hunk 4 above)
Comment (lgodin-nr) [reply alert-policies/dbt-cloud/dbt-test-warning.yml:5]: Upudated

File: alert-policies/dbt-cloud/dbt-resource-failed.yml (hunk 3 above)
Comment (lgodin-nr) [reply alert-policies/dbt-cloud/dbt-resource-failed.yml]: updated

File: data-sources/dbt-cloud/config.yml (hunk 5 above)
Comment (lgodin-nr) [reply data-sources/dbt-cloud/config.yml]: Updated

File: quickstarts/dbt-cloud/config.yml (hunk 9 above)
Comment (lgodin-nr) [reply quickstarts/dbt-cloud/config.yml]: Updated

File: quickstarts/dbt-cloud/config.yml (hunk 11 above)
Comment (lgodin-nr) [reply quickstarts/dbt-cloud/config.yml]: Updated

File: data-sources/dbt-cloud/config.yml (hunk 12 above)
Comment (lgodin-nr) [reply data-sources/dbt-cloud/config.yml]: Updated

File: quickstarts/dbt-cloud/config.yml (hunk 10 above)
Comment (lgodin-nr) [reply quickstarts/dbt-cloud/config.yml]: Updated

**Comment by lgodin-nr** [comment]
@sjyothi54 @mdumpati I updated for all the comments and also updated the documentation links.

File: alert-policies/dbt-cloud/dbt-resource-failed.yml [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+  # Time in seconds; Max value: 172800 (48hrs), null if closeViolationsOnExpiration and openViolationOnExpiration are both 'false'
+  expirationDuration: 259200 
+
+# Advanced Signal Settings
+# Duration of the time window used to evaluate the NRQL Condition
+signal:
+  # How long we wait for data that belongs in each aggregation window 
+  aggregationDelay: null # seconds
+  # The method that determines when we consider an aggregation window to complete so that we can evaluate the signals for violations.
+  aggregationMethod: EVENT_FLOW
```
Comment (sjyothi54) [inline alert-policies/dbt-cloud/dbt-resource-failed.yml:46]:
```suggestion
  aggregationMethod: EVENT_FLOW | EVENT_TIMER
```

File: alert-policies/dbt-cloud/dbt-test-warning.yml [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+  # Time in seconds; Max value: 172800 (48hrs), null if closeViolationsOnExpiration and openViolationOnExpiration are both 'false'
+  expirationDuration: 259200 
+
+# Advanced Signal Settings
+# Duration of the time window used to evaluate the NRQL Condition
+signal:
+  # How long we wait for data that belongs in each aggregation window 
+  aggregationDelay: null # seconds
+  # The method that determines when we consider an aggregation window to complete so that we can evaluate the signals for violations.
+  aggregationMethod: EVENT_FLOW
```
Comment (sjyothi54) [inline alert-policies/dbt-cloud/dbt-test-warning.yml:45]:
```suggestion
  aggregationMethod: EVENT_FLOW | EVENT_TIMER
```

File: data-sources/dbt-cloud/config.yml [hunk 15]
```diff
@@ -0,0 +1,21 @@
+id: dbt-cloud 
+displayName: dbt Cloud 
+description: |
+  Integrates dbt Cloud with New Relic using an Airflow DAG. With our dashboard and alerts, you can easily find problems
+  with dbt Cloud runs and resources (models, seeds, snapshots and tests). Also includes the ability to collect failed 
+  test rows from dbt tests run in Snowflake.
```
Comment (sjyothi54) [inline data-sources/dbt-cloud/config.yml:6]:
```suggestion
Integrates dbt Cloud with New Relic using an Airflow DAG. With our dashboard and alerts, you can easily find problems with dbt Cloud runs and resources (models, seeds, snapshots and tests). Also includes the ability to collect failed test rows from dbt tests run in Snowflake. 
```

File: quickstarts/dbt-cloud/config.yml [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+slug: dbt-cloud-integration 
+
+# Displayed in the UI (required)
+title: dbt Cloud Integration 
+
+# Long-form description of the quickstart (required)
+description: |
+  Integrates dbt Cloud with New Relic using an Airflow DAG. With our dashboard and alerts, you can easily find problems
+  with dbt Cloud runs and resources (models, seeds, snapshots and tests). Also includes the ability to collect dispaly 
+  and alert on test rows from dbt tests run in Snowflake.
```
Comment (sjyothi54) [inline quickstarts/dbt-cloud/config.yml:11]:
```suggestion
Integrates dbt Cloud with New Relic using an Airflow DAG. With our dashboard and alerts, you can easily find problems with dbt Cloud runs and resources (models, seeds, snapshots and tests). Also includes the ability to collect dispaly and alert on test rows from dbt tests run in Snowflake. 
```

File: quickstarts/dbt-cloud/config.yml [hunk 17]
```diff
@@ -0,0 +1,50 @@
...
+authors:
+  - Leo Godin 
+
+# Keywords for filtering / searching criteria in the UI
+keywords:
+  - dbt cloud 
+  - dbt
+
+dataSourceIds:
+  - dbt-cloud
```
Comment (sjyothi54) [inline quickstarts/dbt-cloud/config.yml:29]:
```suggestion
  - dbt-cloud
```

**Review by sjyothi54 — CHANGES_REQUESTED** [review]
@lgodin-nr I checked the failed validations and made necessary request changes

File: alert-policies/dbt-cloud/dbt-failed-test-row.yml [hunk 18]
```diff
@@ -0,0 +1,63 @@
...
+  # Time in seconds; Max value: 172800 (48hrs), null if closeViolationsOnExpiration and openViolationOnExpiration are both 'false'
+  expirationDuration: 259200 
+
+# Advanced Signal Settings
+# Duration of the time window used to evaluate the NRQL Condition
+signal:
+  # How long we wait for data that belongs in each aggregation window 
+  aggregationDelay: null # seconds
+  # The method that determines when we consider an aggregation window to complete so that we can evaluate the signals for violations.
+  aggregationMethod: EVENT_FLOW
```
Comment (sjyothi54) [inline alert-policies/dbt-cloud/dbt-failed-test-row.yml:46]:
```suggestion
  aggregationMethod: EVENT_FLOW | EVENT_TIMER
```

File: alert-policies/dbt-cloud/dbt-job-failed.yml [hunk 19]
```diff
@@ -0,0 +1,62 @@
...
+  # Time in seconds; Max value: 172800 (48hrs), null if closeViolationsOnExpiration and openViolationOnExpiration are both 'false'
+  expirationDuration: 259200 
+
+# Advanced Signal Settings
+# Duration of the time window used to evaluate the NRQL Condition
+signal:
+  # How long we wait for data that belongs in each aggregation window 
+  aggregationDelay: null # seconds
+  # The method that determines when we consider an aggregation window to complete so that we can evaluate the signals for violations.
+  aggregationMethod: EVENT_FLOW
```
Comment (sjyothi54) [inline alert-policies/dbt-cloud/dbt-job-failed.yml:45]:
```suggestion
  aggregationMethod: EVENT_FLOW | EVENT_TIMER
```

File: quickstarts/dbt-cloud/config.yml [hunk 20]
```diff
@@ -0,0 +1,48 @@
+# Sets the URL name of the quickstart on public I/O (required)
+slug: dbt-cloud-integration 
+
+# Displayed in the UI (required)
+title: dbt Cloud Integration 
+
+# Long-form description of the quickstart (required)
+description: |
+Integrates dbt Cloud with New Relic using an Airflow DAG. With our dashboard and alerts, you can easily find problems with dbt Cloud runs and resources (models, seeds, snapshots and tests). Also includes the ability to collect dispaly and alert on test rows from dbt tests run in Snowflake.
```
Comment (sjyothi54) [inline quickstarts/dbt-cloud/config.yml:9]:
```suggestion
  Integrates dbt Cloud with New Relic using an Airflow DAG. With our dashboard and alerts, you can easily find problems with dbt Cloud runs and resources (models, seeds, snapshots and tests). Also includes the ability to collect dispaly and alert on test rows from dbt tests run in Snowflake. 
```

**Review by sjyothi54 — CHANGES_REQUESTED** [review]
@lgodin-nr Can you work on this comments as well

**Comment by sjyothi54** [comment]
Raised new PR for dbt cloud integration:https://github.com/newrelic/newrelic-quickstarts/pull/2456, hence closing this PR

**Pull request closed without merging**